src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1104,7 +1104,7 @@ string PrintProcesses()
                     builder.AppendLine();
                 }
                 
-                builder.AppendFormat("Current process id: {0}", Process.GetCurrentProcess().Id);
+                builder.AppendFormat("Current process id: {0} Process name: '{1}'", Process.GetCurrentProcess().Id, currentProcess.ProcessName);
```

Comment:
Doesn't `currentProcess === Process.GetCurrentProcess()` here? If so why pass in `currentProcess` ?

Reply:
> Doesn't currentProcess === Process.GetCurrentProcess() here? If so why pass in currentProcess ?

I agree that should be...but since the pid is on the list and asserts fail I want to be sure to log the same name used in `Process.GetProcessesByName(currentProcess.ProcessName);` which result that fail in the assert(afaik at this point Process.GetCurrentProcess().ProcessName could be different if called in different time) , I mean I understand that could be unuseful and weid...but also the fail at the moment seems weird to me, among the things I used this apis a lot of time without problem. Finally I'm speculating, this behaviour is very strange, maybe OS api bug?If process name will be ok in log I'll check code row by row to try to understand.